modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -124,6 +124,57 @@ class ONNXImporter
     typedef std::map<std::string, LayerInfo>::iterator IterLayerId_t;
 
     void handleNode(const opencv_onnx::NodeProto& node_proto);
+
+private:
+    typedef void (ONNXImporter::*ONNXImporterNodeParser)(LayerParams& layerParams, opencv_onnx::NodeProto node_proto);
+    typedef std::map<std::string, ONNXImporterNodeParser> DispatchMap;
+
+    void parseMaxPool              (LayerParams& layerParams, opencv_onnx::NodeProto node_proto);
```

Comment:
> opencv_onnx::NodeProto

"const reference" to avoid unnecessary objects copying.

Only some of handlers need local own instances to provide modifications (these code parts looks bad).